### PR TITLE
CircleCI: Implement new release pipeline

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Implement new release pipeline by adapting the "build-pipeline" workflow.

### CODE REVIEW

**Review by xlson — COMMENTED** [review]
Looks good to me, there was nothing obvious that I could see that didn't make sense.

Is the release publisher baked into grabpl or still used btw?

**Comment by aknuds1** [comment]
@xlson Thanks! The release publisher is still standalone, although I'm sure it could be baked into grabpl in the future.

File: .circleci/config.yml [hunk 1]
```diff
@@ -527,129 +428,22 @@ jobs:
           name: CI job started
           command: ./scripts/ci-job-started.sh
       - run: docker info
+      # XXX: Is this necessary?
```
[the remark this answers is not in the document]
Comment (aknuds1) [reply .circleci/config.yml]: @xlson The XXX is for the next line :) `docker run --privileged linuxkit/binfmt:v0.6`.

**Comment by aknuds1** [comment]
Still working out some bugs.

**Comment by marefr** [comment]
@aknuds1 Looked up a rather old commit in master (before your build pipeline changes) and the 6.2.2 release build:
[Old master build](https://circleci.com/workflow-run/358b8d17-5e1a-4194-b747-c61033205d8a?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link) vs [6.2.2 build](https://circleci.com/workflow-run/cd4993f1-4873-4aaf-844c-a0983f58d996?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link)

Looking at your [latest PR build](https://circleci.com/workflow-run/7b7892b6-5198-4a54-9df3-94c28517782f?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link) there's really hard to understand which steps depends on which ones and so on :smile: 

Using their new UI a little bit easier, but more messy :smile: 
[Latest PR build](https://app.circleci.com/pipelines/github/grafana/grafana/3280/workflows/f6f70dca-930a-488e-af9a-381e076421c5)
[6.2.2 build](https://app.circleci.com/pipelines/github/grafana/grafana/2420/workflows/cd4993f1-4873-4aaf-844c-a0983f58d996)
old master build workflow didn't work with the new UI.

Looking at lint-go step for example seems like test-backend and integration tests are dependent steps, but not all of the other "build-*" steps - is that expected? 

Doesn't look right that release-packages-test are run in [Latest PR build](https://app.circleci.com/pipelines/github/grafana/grafana/3280/workflows/f6f70dca-930a-488e-af9a-381e076421c5). For 6.2.2 it's called [release-packages](https://app.circleci.com/jobs/github/grafana/grafana/267961)

end-to-end test for 6.6.2 release build using grafana/grafana-dev:v6.6.2 image. Your  [Latest PR build](https://app.circleci.com/pipelines/github/grafana/grafana/3280/workflows/f6f70dca-930a-488e-af9a-381e076421c5) using grafana/grafana-dev:f6f70dca-930a-488e-af9a-381e076421c5 image. Same as the commit. I guess this is expected, just want to verify.

One possible flaw but at the same time possible good thing with the current approach is that enterprise flow can fail, but OSS succeed or the other way around.

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marefr) [reply .circleci/config.yml]: `docker run --privileged linuxkit/binfmt:v0.6` is needed to be able to build arm images as far as I know.

[the remark this answers is not in the document]
Comment (aknuds1) [reply .circleci/config.yml]: @marefr That's what I think it's for too, but I'm not 100% sure if it's necessary any longer.

**Comment by aknuds1** [comment]
@marefr It's a good question if you should let build steps depend on linting, but it will only save time in case linting fails. The pipeline will halt at the packaging step anyway, if linting failed.

The step is only called "release-packages-test" temporarily, since I want to avoid releasing packages while testing the new release pipeline. It should be fixed up before we finalize this PR.

About end-to-end tests using the grafana/grafana-dev:f6f70dca-930a-488e-af9a-381e076421c5, I switched to a solution of using the CircleCI workflow ID as image tag since it's unique and should work for all pipelines.

Good catch about the fact that enterprise can fail and OSS still finish - I'll make an issue to evaluate this at a later point. I'm not sure right now how to best handle this.

**Comment by torkelo** [comment]
Great job, looks really good!

**Review by papagian — COMMENTED** [review]
LGTM; amazing work!

**Comment by aknuds1** [comment]
Thanks @torkelo and @papagian !!

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marefr) [reply .circleci/config.yml]: Why is it not needed? Think using Docker for Mac it's not needed, but Docker on Ubuntu I think it's still needed. Something to investigate later so open an issue for it is my suggestion to get this merged.

[the remark this answers is not in the document]
Comment (aknuds1) [reply .circleci/config.yml]: @marefr Yeah, it's not a blocker for sure :)

**Review by marefr — APPROVED** [review]
LGTM great work!

**Comment by dprokop** [comment]
Awesome work @aknuds1 🚀

**Pull request merged**